icu4c/source/i18n/dtitvfmt.cpp

```diff
@@ -1556,19 +1642,43 @@ DateIntervalFormat::adjustFieldWidth(const UnicodeString& inputSkeleton,
          0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0
     };
 
+    const int8_t PATTERN_CHAR_BASE = 0x41;
+
     DateIntervalInfo::parseSkeleton(inputSkeleton, inputSkeletonFieldWidth);
     DateIntervalInfo::parseSkeleton(bestMatchSkeleton, bestMatchSkeletonFieldWidth);
+    if (suppressDayPeriodField) {
+        adjustedPtn.findAndReplace(UnicodeString(" a"), UnicodeString());
+        adjustedPtn.findAndReplace(UnicodeString(LOW_A), UnicodeString());
+    }
```

Comment:
Hmm.
1. Time patterns can also have a leading 'a' followed by space, e.g. Korean "a h:mm:ss"
2. Should suppressDayPeriodField also suppress b or B? But maybe that is handled below?

Reply:
OK, thanks for the b/B explanation, I understand now.

Reply:
OK, thanks for the B/b explanation, I understand now. And I am happy with the fix for the space-around-a issue.